caddyhttp/httpserver/replacer.go

```diff
@@ -309,6 +309,12 @@ func (r *replacer) getSubstitution(key string) string {
 	case "{dir}":
 		dir, _ := path.Split(r.request.URL.Path)
 		return dir
+	case "{user}":
+		user, _ := r.request.Context().Value(RemoteUserCtxKey).(string)
+		if user == "" {
+			user = "-"
```

Comment:
I disagree @mholt.  `"-"` should always be the default

What @tobya and I have been discussing on the forums is fixing the **username** logging *and* keeping  the {common} and {combined} log formats compliant with the [Common Log Format ](https://en.wikipedia.org/wiki/Common_Log_Format) which is a common format established by NCSA/W3C for many many years.

The standard says: 
> If a field is unused in a particular entry dash "-" marks the omitted field. 

Why stick with Common Log Format (CLF) for Caddy?   Tooling and log analysis.  Many, many tools such as Logstash/Elasticsearch, AWStats, Splunk, etc.   Most of them are out of the box ready to understand CLF

Most of them use this grok pattern or something similar
```
COMMONAPACHELOG %{IPORHOST:clientip} %{USER:ident} %{USER:auth} \[%{HTTPDATE:timestamp}\] "(?:%{WORD:verb} %{NOTSPACE:request}(?: HTTP/%{NUMBER:httpversion})?|%{DATA:rawrequest})" %{NUMBER:response} (?:%{NUMBER:bytes}|-)
COMBINEDAPACHELOG %{COMMONAPACHELOG} %{QS:referrer} %{QS:agent}
```
These lines setup a default `"-"` and allows plugins that perform authentication to pass the username into the Caddy log